include/boost/process/v2/impl/exe.ipp

```diff
@@ -0,0 +1,313 @@
+// Copyright (c) 2022 Klemens D. Morgenstern
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+#ifndef BOOST_PROCESS_V2_IMPL_EXE_IPP
+#define BOOST_PROCESS_V2_IMPL_EXE_IPP
+
+#include <boost/process/v2/detail/config.hpp>
+#include <boost/process/v2/detail/last_error.hpp>
+#include <boost/process/v2/detail/throw_error.hpp>
+#include <boost/process/v2/detail/exe.hpp>
+#include <boost/process/v2/exe.hpp>
+
+#if defined(BOOST_PROCESS_V2_WINDOWS)
+#include <windows.h>
+#else
+#include <climits>
+
+#if (defined(__APPLE__) && defined(__MACH__))
+#include <libproc.h>
+#endif
+
+#if (defined(BOOST_PROCESS_V2_WINDOWS) || defined(__linux__) || defined(__ANDROID__) || defined(__sun))
+#include <cstdlib>
+#endif
+
+#if (defined(__FreeBSD__) || defined(__DragonFly__) || defined(__NetBSD__))
+#include <sys/types.h>
+#include <sys/sysctl.h>
+#endif
+
+#if defined(__OpenBSD__)
+#include <sys/types.h>
+#include <sys/param.h>
+#include <sys/sysctl.h>
+#include <kvm.h>
+#endif
+
+BOOST_PROCESS_V2_BEGIN_NAMESPACE
+
+namespace ext {
+
+#if defined(BOOST_PROCESS_V2_WINDOWS)
+
+boost::filesystem::path exe_path(pid_type pid, error_code & ec)
+{
+    boost::filesystem::path path;
+    if (pid == GetCurrentProcessId()) 
```

Comment:
I wonder if we (1) need this and (2) if that maybe should be it's own function.

Reply:
I explained already, we do need it. MS docs explain Only use the former function for the current process, and it is better to only get foreign process exe names with the latter otherwise you might end up with wrong results on occasion. I'll link to the documentation page which claims that on msdn if you can't find it. I will admit it does seem a bit odd.

Reply:
> To retrieve the module name of the current process, use the [GetModuleFileName](https://learn.microsoft.com/en-us/windows/desktop/api/libloaderapi/nf-libloaderapi-getmodulefilenamea) function with a NULL module handle. This is more efficient than calling the GetProcessImageFileName function with a handle to the current process.
> 
> To retrieve the name of the main executable module for a remote process in win32 path format, use the [QueryFullProcessImageName](https://learn.microsoft.com/en-us/windows/desktop/api/winbase/nf-winbase-queryfullprocessimagenamea) function.

https://learn.microsoft.com/en-us/windows/win32/api/psapi/nf-psapi-getprocessimagefilenamea